front/components/workspace/ProviderManagementModal.tsx

```diff
@@ -163,6 +178,36 @@ export function ProviderManagementModal({
           })}
         </ContextItem.List>
       </div>
+      <div className="flex flex-row items-center gap-4 px-4 pt-4">
+        <div className="s-text-sm font-semibold">Embedding Provider:</div>
+        <DropdownMenu>
+          <DropdownMenu.Button>
+            <Button
+              type="select"
+              labelVisible={true}
+              label={
+                embeddingProvider
+                  ? prettyfiedProviderNames[embeddingProvider]
+                  : prettyfiedProviderNames["openai"]
+              }
+              variant="secondary"
+              hasMagnifying={false}
+              size="sm"
+            />
+          </DropdownMenu.Button>
+          <DropdownMenu.Items origin="topRight">
+            {EMBEDDING_PROVIDER_IDS.map((provider) => (
```

Comment:
OOC, how do we plan to handle this downstream? Today, we can't just change on the fly, it requires some manual intervention to migrate from one collection to the other!

Reply:
Can we add a warning at least to surface that this changes will only impact the newly created data sources and provide a count of the data sources that are still running on the previous embedder.

Reply:
Will add a warning message to make sure this is understood by the user.

I would personally disable that dropdown with a message "contact us if you want to modify your embedding provider". Otherwise they would indeed have data sources running on different collections

Reply:
@spolu Are we okay with letting them run data sources on different collections and add a warning message?

Reply:
Yes we're good with it technically but we need to probably work a little bit more on the "product" to support switching (eg show that there exist data sources that are not on the default embedding provider)

Proposal: Let's make it a read-only information for now with a hover that invite them to contact us if they want to change it :+1: